sdk/formrecognizer/azure-ai-formrecognizer/tests/testcase.py

```diff
@@ -5,51 +5,289 @@
+                actual_field.elements,
+                read_results
             )
-        return self.generate_fake_token()
 
-    def generate_fake_token(self):
-        return FakeTokenCredential()
+    def assertReceiptItemsTransformCorrect(self, items, actual_items, read_results=None):
+        actual = actual_items.value_array
+
+        for r, a in zip(items, actual):
+            self.assertFormFieldTransformCorrect(r.name, a.value_object.get("Name"), read_results)
+            self.assertFormFieldTransformCorrect(r.quantity, a.value_object.get("Quantity"), read_results)
+            self.assertFormFieldTransformCorrect(r.total_price, a.value_object.get("TotalPrice"), read_results)
+            self.assertFormFieldTransformCorrect(r.price, a.value_object.get("Price"), read_results)
+
+    def assertTablesTransformCorrect(self, layout, actual_layout, read_results=None):
+        for table, actual_table in zip(layout, actual_layout):
+            self.assertEqual(table.row_count, actual_table.rows)
+            self.assertEqual(table.column_count, actual_table.columns)
+            for cell, actual_cell in zip(table.cells, actual_table.cells):
+                self.assertEqual(cell.text, actual_cell.text)
+                self.assertEqual(cell.row_index, actual_cell.row_index)
+                self.assertEqual(cell.column_index, actual_cell.column_index)
+                self.assertEqual(cell.row_span, actual_cell.row_span if actual_cell.row_span is not None else 1)
+                self.assertEqual(cell.column_span, actual_cell.column_span if actual_cell.column_span is not None else 1)
+                self.assertEqual(cell.confidence, actual_cell.confidence if actual_cell.confidence is not None else 1.0)
+                self.assertEqual(cell.is_header, actual_cell.is_header if actual_cell.is_header is not None else False)
+                self.assertEqual(cell.is_footer, actual_cell.is_footer if actual_cell.is_footer is not None else False)
+                self.assertBoundingBoxTransformCorrect(cell.bounding_box, actual_cell.bounding_box)
+                self.assertTextContentTransformCorrect(cell.text_content, actual_cell.elements, read_results)
+
+    def assertReceiptItemsHasValues(self, items, page_number, include_text_content):
+        for item in items:
+            self.assertBoundingBoxHasPoints(item.name.value_data.bounding_box)
+            self.assertIsNotNone(item.name.confidence)
+            self.assertIsNotNone(item.name.value_data.text)
+            self.assertBoundingBoxHasPoints(item.quantity.value_data.bounding_box)
+            self.assertIsNotNone(item.quantity.confidence)
+            self.assertIsNotNone(item.quantity.value_data.text)
+            self.assertBoundingBoxHasPoints(item.total_price.value_data.bounding_box)
+            self.assertIsNotNone(item.total_price.confidence)
+            self.assertIsNotNone(item.total_price.value_data.text)
+
+            if include_text_content:
```

Comment:
you can also assert that if not include_text_content, these properties are None

Reply:
done